src/main/java/com/ibm/darpc/DaRPCEndpointGroup.java

```diff
@@ -38,18 +38,20 @@
 	private int timeout;
 	private int bufferSize;
 	private int maxInline;
+	private DaRPCMemPool memPool;
 	
 	public static int getVersion(){
 		return DARPC_VERSION;
 	}	
 	
-	protected DaRPCEndpointGroup(DaRPCProtocol<R,T> protocol, int timeout, int maxinline, int recvQueue, int sendQueue) throws Exception {
+	protected DaRPCEndpointGroup(DaRPCProtocol<R,T> protocol, DaRPCMemPool memPool, int timeout, int maxinline, int recvQueue, int sendQueue) throws Exception {
```

Comment:
The part I'm probably missing here, but why don't we have a simple interface defining a DaRPCMemPool, and then have applications pass any implementation of that interface while also providing a default implementation within DaRPC. Currently it looks like DaRPCMemPool is both the new MemPool type and the only implementation. Or I'm I missing something?

Reply:
I agree. I changed it. There is an interface now and the application can instantiate a specific implementation of this interface.